src/export/bops/index.ts

```diff
@@ -110,52 +85,55 @@ function exhaustiveCheck(type: never): never {
   throw new Error(`Unhandled type ${type}`);
 }
 
-// For a given node (a "Question"), recursively scan the flow schema to find which portal it belongs to
-//   and add the portal_name to the QuestionMetadata so BOPS can group proposal_details
-const addPortalName = (
-  id: string,
-  flow: LooseFlowGraph,
-  metadata: QuestionMetaData
-): QuestionMetaData => {
-  if (id === "_root") {
-    metadata.portal_name = "_root";
-  } else if (flow[id]?.type === ComponentType.InternalPortal) {
-    // internal & external portals are both type 300 after flattening (ref dataMergedHotFix)
-    metadata.portal_name = flow[id]?.data?.text || id;
-  } else {
-    // if the current node id is not the root or a portal, then find its' next parent node and so on until we hit a portal
-    Object.entries(flow).forEach(([nodeId, node]) => {
-      if (node.edges?.includes(id)) {
-        addPortalName(nodeId, flow, metadata);
-      }
-    });
-  }
+export function formatProposalDetails({
+  flow,
+  breadcrumbs,
+}: {
+  flow: LooseFlowGraph;
+  breadcrumbs: LooseBreadcrumbs;
+}) {
+  const sortedFlow = sortFlow(flow);
+  const hasSections = sortedFlow.some(
+    (node) => node.type === ComponentType.Section
+  );
+  const addSectionName = (
+    id: string,
+    metadata: QuestionMetaData
+  ): QuestionMetaData => {
+    const currentNode = sortedFlow.find((node) => node.id === id);
+    const currentSection = sortedFlow.find(
+      (node) => node.id == currentNode!.sectionId
+    );
+    metadata.section_name = currentSection?.data?.title as string;
+    return metadata;
+  };
```

Comment:
I've inlined these closures based on my initial hunch that this could improve performance but I'm not sure that's actually the case.